src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+
+export const config: Feature = { // An object that describes new feature we introduce
+    id: 'filter_search',  // Feature id - any unique string would do
+    name: 'Filter Search',  // Feature name - would be displayed in the settings menu
+    //warning: 'Experimental feature',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    const jsDoc = document;
+    if (active) {
+        jsDoc.addEventListener("click", filterSearch);
+    } else {
+        jsDoc.removeEventListener("click", filterSearch);
+    }
+}
+
+var debugMode = 0;
+
+function filterSearch(evt: MouseEvent)
+{
+    var evtTarget = (<HTMLElement>evt.target);
+    if(evtTarget !== null)
+    {
+        if(debugMode == 1){console.log(evtTarget.className);}
+
+        if(evtTarget.className === 'bp3-button')
+        {
+            var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+            if(tbInputSearch !== null)
+            {
+                tbInputSearch.focus();
+                tbInputSearch.select();
+            }
+        }
+        else if(evtTarget.className === 'bp3-icon bp3-icon-filter' || evtTarget.className === 'bp3-button bp3-minimal bp3-small')
+        {
+            var waitCtr = 0;
+            function waitForFilter()
+            {
+                setTimeout(function(){
```

Comment:
> make this function async and use [delay](https://github.com/roam-unofficial/roam-toolkit/blob/master/src/ts/core/common/async.ts#L1) instead of setting timeout

Why? What is purpose of this change? Also would need some help/guidance because my code works this way and I don’t necessarily have a strong handle on async stuff. Would need you to help me here.

Reply:
`async/await` is simpler to interact with as the code becomes more linear. would be happy to help with figuring it out. that said, I think if you use https://github.com/uzairfarooq/arrive  as I recommend below you won't need either timer or delay